pkg/internal/ebpf/httpfltr/httpfltr.go

```diff
@@ -124,6 +122,15 @@ func (p *Tracer) Load() (*ebpf.CollectionSpec, error) {
 		loader = loadBpf_debug
 	}
 
+	kernelMajor, kernelMinor := ebpf2.KernelVersion()
+	if kernelMajor > 5 || (kernelMajor == 5 && kernelMinor >= 17) {
+		p.log.Info("Found Linux kernel later than 5.17, enabling trace information parsing", "major", kernelMajor, "minor", kernelMinor)
+		loader = loadBpf_tp
+		if p.cfg.EBPF.BpfDebug {
+			loader = loadBpf_tp_debug
+		}
+	}
+
```

Comment:
Nice!

Reply:
Will this mean that 1.0 users getting Traceparent with kernel versions < 5.17 will stop getting it?

Should we enclose all this code in an `if config.TrackRequestHeaders {` ?

Reply:
Yes, good point. I'm not sure what to do there, there's no chance we'll ever be able to implement propagation of traces with userspace parsing of the traceparent, and passing along all of the headers caused a lot of overhead.

I'll implement this suggestion with the config option, it helps us test both paths in Github without having to use VMs.

One other option is to implement an alternative `bpf_strstr` with a regular unrolled loop. It won't be able to go for long, but we might have partial support for traceparent if the header is seen in the first 100 bytes or so.